packages/calypso-e2e/src/lib/pages/plans-page.ts

```diff
@@ -17,31 +22,41 @@ const selectors = {
 	activeNavigationTab: ( tabName: PlansPageTab ) =>
 		`.is-selected.section-nav-tab:has-text("${ tabName }")`,
 
-	actionButton: ( { plan, buttonText }: { plan: Plan; buttonText: PlanActionButton } ) => {
+	actionButton: ( { plan, buttonText }: { plan: LegacyPlans; buttonText: PlanActionButton } ) => {
 		const viewportSuffix = envVariables.VIEWPORT_NAME === 'mobile' ? 'mobile' : 'table';
 		return `.plan-features__${ viewportSuffix } >> .plan-features__actions-button.is-${ plan.toLowerCase() }-plan:has-text("${ buttonText }")`;
 	},
 
-	// Plans view
-	plansGrid: '.plans-features-main',
+	// Legacy plans view
+	legacyPlansGrid: '.plans-features-main',
+
+	// Overhauled plans view
+	overhauledPlansGrid: 'div.plans > table',
+	upgradeToProButton: 'th button.is-primary',
+	planComparisonActionButton: ( action: PlansComparisonAction ) => {
+		const buttonText = `${ toTitleCase( action ) } full plan comparison`;
+		return `button:text("${ buttonText }")`;
+	},
 
 	// My Plans view
-	myPlanTitle: ( planName: Plan ) => `.my-plan-card__title:has-text("${ planName }")`,
+	myPlanTitle: ( planName: LegacyPlans ) => `.my-plan-card__title:has-text("${ planName }")`,
 };
 
 /**
  * Page representing the Plans page accessible at Upgrades >> Plans
  */
 export class PlansPage {
 	private page: Page;
+	private version: PlansGridVersion;
 
 	/**
 	 * Constructs an instance of the Plans POM.
 	 *
 	 * @param {Page} page Instance of the Playwright page
 	 */
-	constructor( page: Page ) {
+	constructor( page: Page, version: PlansGridVersion ) {
```

Comment:
I looked at keying off something else here, but the two-week ago me did not find anything useful to key off of.

The problem is partially due to the slower loading times for the Legacy plans grid which, frustratingly, is **not caught** by the `waitForLoadState('networkidle')`. 

Given the above, I tried to implement a `try/catch` to first look for the new plan, and fall back to the Legacy plan, but that introduces new problems:
- adds execution time if the page being shown is the legacy plans grid.
- what if the overhauled plans page takes just a bit longer than expected to load?

With all things considered, I felt it was important to require, explicitly, that the test writer specify what version of the Plans page they are intending to work with.

Reply:
>With all things considered, I felt it was important to require, explicitly, that the test writer specify what version of the Plans page they are intending to work with.

Given our most recent discussion on the editor, I think this is a great approach! It's very deterministic -- we should know which page we expect. And if we don't get the one we expect, we should probably fail! So I think this will serve us very well 😄